File: backend/src/agent_86/domain/schemas/session_summary.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ActionItem(BaseModel):
    description: str = Field(min_length=1)
    status: Literal["open", "done", "abandoned"]
    owner: str | None = None


class ArtifactRef(BaseModel):
    name: str = Field(min_length=1)
    artifact_type: Literal["docx", "pptx", "xlsx", "diagram", "code", "other"]
    location: str = Field(min_length=1)


class ChatSessionSummary(BaseModel):
    session_id: str = Field(min_length=1)
    title: str = Field(min_length=1)
    date_range_start: datetime
    date_range_end: datetime
    one_line_summary: str = Field(min_length=1)
    topics: list[str] = Field(default_factory=list)
    key_decisions: list[str] = Field(default_factory=list)
    action_items: list[ActionItem] = Field(default_factory=list)
    artifacts_generated: list[ArtifactRef] = Field(default_factory=list)
    open_questions: list[str] = Field(default_factory=list)
    tools_used: list[str] = Field(default_factory=list)
    tags: list[str] = Field(default_factory=list)
    continuation_context: str = Field(default="")


class SessionSummaryResponse(ChatSessionSummary):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/sessionSummary.ts
export interface ActionItem {
  description: string;
  status: "open" | "done" | "abandoned";
  owner: string | null;
}

export interface ArtifactRef {
  name: string;
  artifact_type: "docx" | "pptx" | "xlsx" | "diagram" | "code" | "other";
  location: string;
}

export interface SessionSummary {
  id: string;
  session_id: string;
  user_id: string;
  title: string;
  date_range_start: string;
  date_range_end: string;
  one_line_summary: string;
  topics: string[];
  key_decisions: string[];
  action_items: ActionItem[];
  artifacts_generated: ArtifactRef[];
  open_questions: string[];
  tools_used: string[];
  tags: string[];
  continuation_context: string;
  created_at: string;
  updated_at: string;
}